pkg/trace/agent/agent.go

```diff
@@ -759,3 +772,20 @@ func newEventProcessor(conf *config.AgentConfig, statsd statsd.ClientInterface)
 func (a *Agent) SetGlobalTagsUnsafe(tags map[string]string) {
 	a.conf.GlobalTags = tags
 }
+
+// traceChunkContainsProbabilitySampling returns true when trace has already
+// been sampled by the probabilistic sampler in the OTLPReceiver.
+// The probabilistic sampler in the OTLPReceiver returns `sampler.PriorityAutoKeep`
+// as a result of the sampling decision.
+func traceChunkContainsProbabilitySampling(chunk *pb.TraceChunk) bool {
+	if chunk == nil {
+		return false
+	}
+	if dm, ok := chunk.Tags[tagDecisionMaker]; ok && dm == probabilitySampling {
+		return true
+	}
+	if spans := chunk.GetSpans(); len(spans) > 0 && spans[0].Meta[tagDecisionMaker] == probabilitySampling {
+		return true
+	}
+	return false
+}
```

Comment:
To @ajgajg1134 's https://github.com/DataDog/datadog-agent/pull/33695#discussion_r1953382920

> If I'm wrong about the above then one small different note: Sorry I didn't notice before, but I believe it would be preferred to check the trace chunk for this tag and not just look at the very first span (as we're very slowly trying to move folks to using tags at the right "level").

Added two patterns to check a trace chunk contains dm.